### ISSUE TITLE
Can't login to imap account on startup

### ISSUE THREAD

**Issue by Ajnasz** [neomutt/neomutt#1479]
Since commit a247431dc1d54c8907476af17f6542b295ede9bf my neomutt can't login at
startup to my IMAP email accounts.
It's working if I compile cf90287291386386d0377417257e7eedbe073a37

I have a `account-init` file, where I decrypt the gpg encrypted passwords:
```
# decrypt
set my_tmpsecret="`gpg -o ~/.secret/.tmp -d ~/.secret/.passwd.gpg`"

# store passwords
set my_account1_pass="`awk '/Account1/ {print $2}' ~/.secret/.tmp`"
set my_account2_pass="`awk '/Account2/ {print $2}' ~/.secret/.tmp`"

# remove decrypted file
set my_tmpsecretdel=`rm -f ~/.secret/.tmp`
```

Then I have the following setup:
```
source ~/.mutt/accounts-init

set folder="imaps://me@example.com@mail.example.com/"
mailboxes imaps://me@example.com@mail.example.com/INBOX

set folder="imaps://me@somewhere.com@mail.somewhere.com/"
mailboxes imaps://me@somewhere.com@mail.somewhere.com/INBOX

folder-hook 'me@example.com' 'source ~/.mutt/me@example.com'
account-hook example.com 'set imap_user=me@example.com imap_pass=$my_account1_pass'

folder-hook 'me@somewhere.com' 'source ~/.mutt/me@somewhere.com'
account-hook somewhere.com 'set imap_user=me@somewhere.com imap_pass=$my_account2_pass'
```

Then I load the default account:

```
source ~/.mutt/me@example.com
```

In the folder-hook file ~/.mutt/me@example.com I have setup like this:

```
set folder="imaps://me@example.com@mail.example.com/"
set spoolfile="imaps://me@example.com@mail.example.com/INBOX"
set smtp_url = "smtp://me@example.com@mail.example.com:587" smtp_pass="$my_account1_pass"

set from="me@example.com"
set hostname="example.com"
set record="imaps://me@example.com@mail.example.com/Sent"
set postponed="imaps://me@example.com@mail.example.com/Drafts"
set trash="imaps://me@example.com@mail.example.com/Trash"

my_hdr From: ...
my_hdr Reply-To: ...
```

And I have a similar setup for all of my accounts

Any idea what should I change to make it work again?

**Comment by flatcap** [comment]
tl;dr
Move your `mailboxes` commands after your `account-hook`s and `folder-hook`s.

Currently, IMAP mailboxes are opened when then the `mailboxes` command is processed, so that we can ask the server for the path delimiter.  This behaviour may change.

The current version of the code (from git master) is under heavy development -- we're trying to modernise the code.
While we do, it's not suitable for general use.

If you're keen to build your own NeoMutt, then I recommend the latest stable release: NeoMutt 2018-07-16. `git checkout neomutt-20180716` then clean up and rebuild.

If you like living on the edge, then we welcome inquisitive testers.
We're developing hard and new code is going in daily.

**Comment by Ajnasz** [comment]
Thank you, that helps. I see the progress and thank you for the hard work! Will see if I can help with useful testing. :)
